valves/__init__.py

```diff
@@ -0,0 +1,66 @@
+def sessionize(dataf, user_col="user", ts_col="timestamp", threshold=20 * 60):
+    """
+    Adds a session to the dataset, using multiple dispatch to proxy the correct backend.
+
+    This function is meant to be used in a `.pipe()`-line.
+
+    Arguments:
+        - dataf: pandas, polars or dask dataframe
+        - user_col: name of the column containing the user id
+        - ts_col: name of the column containing the timestamp
+        - threshold: time in seconds to consider a user inactive
+    """
+    if "dask" in str(dataf.__class__):
```

Comment:
I think we can just do type matching? 

`if _POLARS_AVAILABLE and isinstance(dataf, pl.DataFrame)`

`if _PANDAS_AVAILABLE and isinstance(dataf, pd.DataFrame) `

The `globals` are only needed if we want to support this library without the need to have all dataframe libraries installed.

They could then be set on import.

```python
try:
   import polars as pl
   _POLARS_AVAILABLE = True
except ImportError:
    _POLARS_AVAILABLE = False
```

In polars we do something similar with `pyarrow`

Reply:
Due to shortcircuiting the `isinstance(dataf, pd.DataFrame)` will not run if pandas is not installed. I think this would work.

Reply:
Fair enough. The `isinstance` certainly is less janky than a string match.